source/Camera/InfoUtilOther/ZividBenchmark/ZividBenchmark.cpp

```diff
@@ -459,6 +472,71 @@ namespace
         printCapture2DResults(allDurations);
     }
 
+    Zivid::Frame assistedCapture(Zivid::Camera &camera)
+    {
+        const auto parameters = Zivid::CaptureAssistant::SuggestSettingsParameters{
+            Zivid::CaptureAssistant::SuggestSettingsParameters::AmbientLightFrequency::none,
+            Zivid::CaptureAssistant::SuggestSettingsParameters::MaxCaptureTime{ std::chrono::milliseconds{ 800 } }
+        };
+        const auto settings = Zivid::CaptureAssistant::suggestSettings(camera, parameters);
+        return camera.capture(settings);
+    }
+
+    template<typename DataType>
+    Duration copyDataTime(Zivid::Frame frame, DataType copyName)
```

Comment:
You can simplify this by removing argument `DataType copyName` here, replace `decltype(copyName)` with just `DataType`, and then below you invoke it like `copyDataTime<Zivid::PointXYZ>(frame);`

Reply:
The point is to use `Zivid::PointXYZ` or `Zivid::PointXYZW` or any other, not only `Zivid::PointXYZ`. How you you achieve that?